Sources/SKCore/FileBuildSettingsChange.swift

```diff
@@ -16,8 +16,16 @@ public enum FileBuildSettingsChange {
   /// The `BuildSystem` has no `FileBuildSettings` for the file.
   case removedOrUnavailable
 
-  /// The `FileBuildSettings` have been modified.
+  /// The `FileBuildSettings` have been modified or are newly available.
   case modified(FileBuildSettings)
+
+  /// The `BuildSystem` is providing fallback arguments which may not be correct,
+  /// or potentially is missing some generated inputs (e.g. prebuilding required).
+  ///
+  /// This indicates that diagnostics should be withheld for the given file. In addition, going from
+  /// fallback --> modified with the exact same `FileBuildSettings` is supported and indicates that
+  /// the LSP should rebuild its AST for the given file and emit new diagnostics.
```

Comment:
Why would you need to rebuild anyway, isn't it just a question of no longer filtering?

Reply:
I think this would only happen in practice if the build system does it on purpose - from above - to note that generated inputs are now available. How would fallback --> primary happen with the same settings otherwise?

Maybe the documentUpdatedDependencies should handle this instead somehow? Unclear to me how that would work, let me know if you have ideas.